Cython/Compiler/ModuleNode.py

```diff
@@ -233,8 +233,29 @@ def h_entries(entries, api=0, pxd=0):
                 h_code.putln('#error "Unicode module names are not supported in Python 2";')
             h_code.putln("PyMODINIT_FUNC init%s(void);" % py2_mod_name)
             h_code.putln("#else")
-            h_code.putln("PyMODINIT_FUNC %s(void);" % self.mod_init_func_cname('PyInit', env))
-            h_code.putln("#endif")
+            py3_mod_func_name = self.mod_init_func_cname('PyInit', env)
+            warning_string = EncodedString(
+                'Use PyImport_AppendInittab("%s", %s) instead of calling %s directly.' % (
+                                    py2_mod_name, py3_mod_func_name, py3_mod_func_name))
+            h_code.putln('/* WARNING: From Python 3.5 %s */' % warning_string)
+            h_code.putln("PyMODINIT_FUNC %s(void);" % py3_mod_func_name)
+            h_code.putln("")
+            h_code.putln("#if PY_MINOR_VERSION >= 5 && (defined(__GNUC__) || defined(__clang__) || defined(_MSC_VER))")
```

Comment:
I would like to see a full Python hex version comparison here. Otherwise, this will silently be discarded in Py4.0.